numpy/random/mtrand/mtrand.pyx

```diff
@@ -641,8 +650,17 @@ cdef class RandomState:
         RandomState
 
         """
+        if version is None:
+            version = HIGHEST_VERSION if seed is None else 0
+        if version not in range(HIGHEST_VERSION + 1):
+            raise ValueError("`version` must be an integer between 0 and {}".
+                             format(HIGHEST_VERSION))
+        self.version = version
+
         cdef rk_error errcode
         cdef ndarray obj "arrayObject_obj"
+        self._free_internal_state()
+        self.internal_state = <rk_state*>PyMem_Malloc(sizeof(rk_state))
```

Comment:
Yes, this is a leftover of the initial attempt at supporting various internal PRNGs as well (in which case the internal state needs to be reallocated whenever the underlying PRNG is changed).  I'll get rid of it.